hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenAirdropTest.java

```diff
@@ -2238,5 +2248,228 @@ private HapiTokenCreate createNft(String tokenName, String supplyKey) {
+        }
+
+        @HapiTest
+        @DisplayName("when token is frozen")
+        final Stream<DynamicTest> whenTokenIsFrozen() {
+            final String ALICE = "alice";
+            var mutableContract = "PayReceivable";
+            final String FUNGIBLE_TOKEN_A = "fungibleTokenA";
+            return hapiTest(flattened(
+                    newKeyNamed("freezeKey"),
+                    cryptoCreate(ALICE).balance(ONE_HBAR),
+                    deployMutableContract(mutableContract, 2),
+                    tokenCreate(FUNGIBLE_TOKEN_A)
+                            .treasury(ALICE)
+                            .tokenType(FUNGIBLE_COMMON)
+                            .freezeKey("freezeKey")
+                            .initialSupply(15L),
+                    tokenFreeze(FUNGIBLE_TOKEN_A, ALICE),
+                    tokenAssociate(mutableContract, FUNGIBLE_TOKEN_A),
+                    tokenAirdrop(moving(10, FUNGIBLE_TOKEN_A).between(ALICE, mutableContract))
+                            .payingWith(ALICE)
+                            .signedByPayerAnd(ALICE)
+                            .hasKnownStatus(ACCOUNT_FROZEN_FOR_TOKEN)));
+        }
+
+        @HapiTest
+        @DisplayName("when airdrop to not associated contract with no free associations - crypto transfer should fail")
+        final Stream<DynamicTest> airdropToNotAssociatedContractWithNoFreeAssociations() {
+            var mutableContract = "PayReceivable";
+            return hapiTest(flattened(
+                    deployMutableContract(mutableContract, 0),
+                    tokenAirdrop(moving(1, FUNGIBLE_TOKEN).between(OWNER, mutableContract))
+                            .payingWith(OWNER),
+                    cryptoTransfer(moving(1, FUNGIBLE_TOKEN).between(OWNER, mutableContract))
+                            .payingWith(OWNER)
+                            .hasKnownStatus(TOKEN_NOT_ASSOCIATED_TO_ACCOUNT)));
+        }
+
+        @HapiTest
+        @LeakyHapiTest(
+                requirement = PROPERTY_OVERRIDES,
+                overrides = {"entities.unlimitedAutoAssociationsEnabled"})
+        @DisplayName("airdrop NFT to hollow account remains when we deploy a contract on it's address")
+        final Stream<DynamicTest> nftToHollowAccountRemainsOnCreate2() {
+            final var contract = "Create2Factory";
+            final var adminKey = "adminKey";
+            final var salt = BigInteger.valueOf(42);
+            final AtomicReference<String> factoryEvmAddress = new AtomicReference<>();
+            final AtomicReference<String> expectedCreate2Address = new AtomicReference<>();
+            final AtomicReference<byte[]> testContractInitcode = new AtomicReference<>();
+            return hapiTest(flattened(
+                    // turning this off so when we create the contract it's with maxAutoAssociation value of 0
+                    overriding("entities.unlimitedAutoAssociationsEnabled", "false"),
```

Comment:
We should move all tests that are setting  `entities.unlimitedAutoAssociationsEnabled=false` to a different suite with `overridesInClass()`. This would also allow using the DSL to create the contract and avoid all those AtomicReferences